AZ3166/src/libraries/MQTT/src/MQTTNetwork.h

```diff
@@ -9,46 +9,92 @@
 #include "SystemWiFi.h"
 #include "Telemetry.h"
 
-class MQTTNetwork {
-public:
-    MQTTNetwork() {
-        socket = new TCPSocket();
+class MQTTNetwork
+{
+  public:
+    MQTTNetwork()
+    {
+        _tcpSocket = NULL;
     }
 
-    ~MQTTNetwork() {
-        delete socket;
+    ~MQTTNetwork()
+    {
+        if (_tcpSocket != NULL)
+        {
+            _tcpSocket->close();
+            delete _tcpSocket;
+            _tcpSocket = NULL;
+        }
     }
 
-    int read(unsigned char* buffer, int len, int timeout) {
-        return socket->recv(buffer, len);
+    int read(unsigned char *buffer, int len, int timeout)
+    {
+        return _tcpSocket->recv(buffer, len);
     }
 
-    int write(unsigned char* buffer, int len, int timeout) {
-        return socket->send(buffer, len);
+    int write(unsigned char *buffer, int len, int timeout)
+    {
+        return _tcpSocket->send(buffer, len);
     }
 
-    int connect(const char* hostname, int port) {
-        int ret;
-        if ( (ret = socket->open(WiFiInterface())) != 0) {
+    int connect(const char *hostname, int port)
+    {
+        if (_tcpSocket != NULL)
+        {
+            return NSAPI_ERROR_OK;
+        }
+        else
+        {
+            _tcpSocket = new TCPSocket();
+            if (_tcpSocket == NULL)
+            {
+                return NSAPI_ERROR_NO_SOCKET;
+            }
+
+            int ret;
+
+            if ((ret = _tcpSocket->open(WiFiInterface())) == 0 && (ret = _tcpSocket->connect(hostname, port)) == 0)
+            {
+                char telemetry[128];
+                snprintf(telemetry, 128, "ret: %d, host: %s", ret, hostname);
+
```

Comment:
remove host %s  hostname here for GDPR and then change telemetry size from 128 to 16